doc/chapters/api/index.rst

```diff
@@ -32,102 +32,187 @@ BindMount
+    * LXC could not destroy the container
 
-|
+Failing to destroy the container leads to it being put into an invalid state. One should
+(ironically) destroy such a container.
 
 .. _dbus-execute:
 
 Execute
 -------
 Launches the specified application/code in the container.
 
-:Parameters:
-        :containerID: ``int32`` The ID obtained by CreateContainer method.
-        :commandLine: ``string`` the method to run in container.
-        :workDirectory: ``string`` path to working directory.
-        :outputFile: ``string`` output file to direct stdout.
-        :env: ``map<string, string>`` environment variables and their values.
-
-|
-
-:Return Values:
-        :pid: ``int32`` PID of process running in the container, as seen by the host.
+Parameters
+##########
+* containerID: ``int32`` The ID obtained by CreateContainer method.
+* commandLine: ``string`` the method to run in container.
+* workDirectory: ``string`` path to working directory.
+* outputFile: ``string`` output file to direct stdout.
+* env: ``map<string, string>`` environment variables and their values.
+
+Return value
+############
+* pid: ``int32`` PID of process running in the container, as seen by the host.
+
+Error sources
+#############
+* Bad state: The container is not ready or is suspended
+* Capability error: couldn't set default capabilities (if ``SetCapabilities`` has not already been
+  called)
+* LXC error: The call to ``attach()`` failed.
+
+Note that even if the ``Execute`` call works fine, that doesn't mean the command that is being run
+inside the container runs fine. For example, it is possible to pass command-lines that point to
+non-executables, or non-existing files. One would notice this however, by getting a
+``ProcessStateChanged`` signal sent when the call exits.
 
 List
 ----
 Returns a list of the current containers
 
-:Return Value:
-        :containers: ``array<int32>`` IDs for all containers
+Return value
+############
+* containers: ``array<int32>`` IDs for all containers
+
+Error sources
+#############
+None, this method only inspects the current state
 
 ListCapabilities
 ----------------
 Lists all capabilities that the user can apply. Note that this does not include
 the default capabilities, which are not listable.
 
-:Return Value:
-        :capabilities: ``array<string>`` all available capability names
+Return value
+############
+* capabilities: ``array<string>`` all available capability names
+
+Error sources
+#############
+None, this method only inspects the current state
 
 Resume
 ------
 Resumes a suspended container
 
-:Parameters:
-        :containerID: ``int32`` The ID obtained by CreateContainer method.
+Parameters
+##########
+* containerID: ``int32`` The ID obtained by CreateContainer method.
+
+Error sources
+#############
+* Bad state: The container is not suspended
+* LXC error: Couldn't resume the container
 
-|
+Failure to resume a container leads to it being put in an invalid state. One should destroy any such
```

Comment:
Maybe we should mention state in the beginning of the page, e.g what an invalid state is and what that means, as it will be the same in all cases.